lib/index.js

```diff
@@ -127,6 +127,16 @@ const hydrate = (fields, doc, type) => {
 exports.toFields = (schema, doc, errors = []) => {
   const fields = expandFields(schema);
   if (doc) hydrate(fields, doc, 'value');
+
+  if (errors) {
+    errors = errors.reduce((obj, error) => {
+      const errorField = error.extensions.dotPath.replace(/.*\.input\./, '').replace(/\[(\d+)\]/, (_, i) => `.${i}`);
+      return dotProp.set(obj, errorField, error.message);
+    }, {});
```

Comment:
I'd pull the error structure preparation out into it's own function so you're only putting the if condition something like this:

```
if (errors) {
  hydrate(fields, formatErrors(errors), 'error');
}
```